R/format_PEW_temp.R

```diff
@@ -0,0 +1,419 @@
+#'                 everything())
+#'
+#'
+#'
+#'
+#' #### check columns with NAs
+#' # sort(apply(pew_data, 2, function(x) sum(is.na(x))), decreasing = TRUE)
+#'
+#'
+#'
+#' #### --------------------------------------~
+#' #### INDIVIDUAL DATA                    #### done, need to check
+#' #### --------------------------------------~
+#'
+#' #### Still to solve:
+#' # There are some individuals (for example 13617043), where the rows of data related
+#' # to the same individual seem complementary, but are in two rows with
+#' # insufficient information when treating the rows separately.
+#' # How to deal with that?
+#' # >>> So far, I remove duplicated rows at the end of the function,
+#' # but there are some rows still maintained, as there is some difference in one
+#' # of the columns of the dataframe.
+#' #
+#' # There is no data about broodID, so far I used NA.
+#' #
+#' # Remove data with no date in the primary data?
+#'
+#'
+#'
+#' message("Compiling individual information...")
+#'
+#' Individual_data <- create_individual_PEW(pew_data)
+#'   dplyr::filter(Species %in% species)
+#'
+#'
+#' create_individual_PEW <- function(data){
+#'
+#'   Individual_data <-
+#'     data %>%
+#'     #### Remove rows without bird ID
+#'     filter(!is.na(IndvID)) %>%
+#'     #### CHECK: Remove rows without date
+#'     # filter(!is.na(Date)) %>%
+#'     #### Format and create new data columns
+#'     dplyr::select(IndvID, Species, PopID, Age, Method, BreedingSeason, Date, Sex) %>%
+#'     dplyr::rename(RingSeason = BreedingSeason) %>%
+#'     dplyr::mutate(RingAge = dplyr::case_when(Age == "0" & Method == "ChickRinging" ~ "chick",
+#'                                              Age == "1" ~ "adult",
+#'                                              Age == ">=2" ~ "adult"),
+#'                   BroodIDLaid    = NA_character_,
+#'                   BroodIDFledged = NA_character_) %>%
+#'     group_by(IndvID) %>%
+#'     dplyr::mutate(Sex_calculated = purrr::map_chr(.x = list(unique(na.omit(Sex))), .f = ~{
+#'                                                             if(length(..1) == 0){
+#'                                                               return(NA_character_)
+#'                                                             } else if(length(..1) == 1){
+#'                                                               return(..1)
+#'                                                             } else {
+#'                                                               return("C")
+#'                                                             }
+#'                                                           }),
+#'                   Sex_genetic = NA_character_) %>%
+#'     dplyr::ungroup() %>%
+#'     #### Final arrangement
+#'     dplyr::select(IndvID, Species, PopID, BroodIDLaid, BroodIDFledged,
+#'                   RingSeason, RingAge, Sex_calculated, Sex_genetic) %>%
+#'     #### Remove duplicated rows (CHECK)
+#'     dplyr::distinct()
```

Comment:
The Individual_data table should include a *single record for each individual* with information that should remain static throughout their life (e.g. sex). The PEW primary data is structured more like our Capture_data, with *multiple records for each individual* with information that changes for each capture (e.g. mass, tarsus). To achieve the Individual_data output we'll need to group_by()/summarise() for each individual, otherwise we'll end up with multiple records for each individual (e.g. look at 13617015 as an example). I'd recommend run `create_capture_PEW` first and then use this as the basis for the `create_individual_PEW`. 😄